src/rdf4cpp/storage/identifier/NodeBackendID.hpp

```diff
@@ -138,8 +138,9 @@ struct alignas(uint64_t) NodeBackendID {
         return underlying_;
     }
 
-    explicit constexpr operator underlying_type() const noexcept {
-        return underlying_;
+    template<std::unsigned_integral I> requires (sizeof(I) >= sizeof(underlying_type))
+    explicit constexpr operator I() const noexcept {
```

Comment:
I would add something like:
```
static_assert(std::is_same_v<decltype(static_cast<uintptr_t>(std::declval<NodeBackendID>())), uintptr_t>,
              "NodeBackendID should be convertible to uintptr_t");
static_assert(std::is_same_v<decltype(static_cast<uint64_t>(std::declval<NodeBackendID>())), uint64_t>,
              "NodeBackendID should be convertible to uint64_t");
```

after the struct. The idea is to guarantee that it is instantiatable without the test cases. 

Side note: default constructor should be constexpr as well so. Then we can get rid of the `std::declval` above.